src/gov/nist/javax/sip/stack/SIPTransactionStack.java

```diff
@@ -113,6 +113,9 @@
     // A set of methods that result in dialog creations.
     protected static final Set<String> dialogCreatingMethods = new HashSet<String>();
 
+    private transient Set<SIPDialogInjectionListener> dialogInjectionListeners = 
```

Comment:
Since we are not provinding any collision resolution logic (just first it taken), I would prefer to make it predictable, so we always get the same behavior (we always get the same Dialog). Probably a List will make more sense than a Set impl with ordering(since i dont see how listeners could be naturally ordered, unless we incoportae a priority)